src/components/Button/Button.tsx

```diff
@@ -105,8 +106,12 @@ const button = ({
     ? <span className={theme.customIcon}><Icon source="caretDown" /></span>
     : null;
 
+    const titleOrAccesibility = title || accessibilityLabel;
   const iconMarkup = icon
-    ? <span className={theme.customIcon}><Icon source={icon} /></span>
+    ? <span className={theme.customIcon}>
+        <Icon source={icon} />
+        {titleOrAccesibility && <VisuallyHidden>{titleOrAccesibility}</VisuallyHidden>}
```

Comment:
@emgage-smit can you review this logic. It needs to:
1. Check if title or accessibilityLabel exists, and if so, place it within the VisuallyHidden component.
2. If both exists then title should be used
3. If nether exist, then it should bypass adding the VisuallyHidden component

I feel the code has a weird structure, as also does not meet requirements 2 and 3.

Can you please review and provide Farhan some assistance if needed. Thanks